projects/igniteui-angular/src/lib/avatar/avatar.component.ts

```diff
@@ -37,58 +37,202 @@ export enum Size {
     templateUrl: 'avatar.component.html'
 })
 export class IgxAvatarComponent implements OnInit, AfterViewInit {
+
+
+    /**
+     * This is a reference to the avatar image element in the DOM.
+     *
+     * ```typescript
+     *  this.avatar.image.style.backgroundImage = "images/picture.jpg";
+     * ```
+     *
+     * @memberof IgxAvatarComponent
+     */
     @ViewChild('image')
     public image: ElementRef;
 
+
+    /**
+     *@hidden
+     *
+     * @memberof IgxAvatarComponent
+     */
     @ViewChild('imageTemplate', { read: TemplateRef })
     protected imageTemplate: TemplateRef<any>;
 
+    /**
+     *@hidden
+     * @memberof IgxAvatarComponent
+     */
     @ViewChild('initialsTemplate', { read: TemplateRef })
     protected initialsTemplate: TemplateRef<any>;
-
+    /**
+     *@hidden
+     * @memberof IgxAvatarComponent
+     */
     @ViewChild('iconTemplate', { read: TemplateRef })
     protected iconTemplate: TemplateRef<any>;
-
+    /**
+     * Indicates that the avatar's attribute aria-label is 'avatar'.
+     * ```html
+     * <igx-avatar aria-label="avatar"></igx-avatar>
+     * ```
+     * @memberof IgxAvatarComponent
+     */
     @HostBinding('attr.aria-label')
     public ariaLabel = 'avatar';
-
+    /**
+     * Indicates that the avatar's attribute role is 'img'.
+     *
+     * ```html
+     * <igx-avatar role="img"></igx-avatar>
+     * ```
+     *
+     * @memberof IgxAvatarComponent
+     */
     @HostBinding('attr.role')
     public role = 'img';
-
+    /**
+     * Initializes the avatar's class.
```

Comment:
This is only a getter - it returns the avatar's class, does not set it or initialize it. Therefore the code asnippet should be how to get it in typescript.